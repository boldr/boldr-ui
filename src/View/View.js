import styled from 'styled-components';

const View = styled.div`
  margin: 0 auto;
  height: 100%;
  width: 100%;
  min-height: 100vh;
`;
export default View;
