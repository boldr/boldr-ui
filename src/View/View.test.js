import React from 'react';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import View from './View';

describe('<View />', () => {
  const wrapper = shallow(<View />);
  it('renders <View /> without breaking', () => {
    expect(wrapper.find('div').length).toBe(1);
  });
  it('should render CSS rules', () => {
    const tree = renderer.create(<View />).toJSON();
    expect(tree).toHaveStyleRule('width', '100%');
  });

  it('renders snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
