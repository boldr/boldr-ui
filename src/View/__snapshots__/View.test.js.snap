// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View /> renders snapshot 1`] = `
.c0 {
  margin: 0 auto;
  height: 100%;
  width: 100%;
  min-height: 100vh;
}

<div
  className="c0"
/>
`;
